fix: do not filter out rows where PARTITION BY resolves to null (#4823)
confluentinc/ksql: docs-md/developer-guide/joins/partition-data.md

Otherwise, Kafka is likely to interleave messages. The use case will determine
if these ordering guarantees are acceptable.

!!! important

>       If the PARTITION BY expression evaluates to NULL the resulting row will be produced to a

Reviewer: ```suggestion
      If the PARTITION BY expression evaluates to NULL, the resulting row is produced to a
```

      random partition. You many want to use [COALESCE](../syntax-reference#coalesce) to wrap

>       the expression convert any NULL values to a default value, e.g.

Reviewer: ```suggestion
      the expression and convert any NULL values to a default value, for example,
```
